aip/client-libraries/4236.md

```diff
@@ -0,0 +1,33 @@
+---
+id: 4236
+state: reviewing
+created: 2024-05-16
+---
+
+# Version-aware clients
+
+APIs can annotate services with [`google.api.api_version`][]. If
+`google.api.api_version` is specified, version-aware clients **must**
+include the value of `google.api.api_version` in the request to the API.
+
+### Expected Generator and Client Library Behavior
+
+If a service is annotated with `google.api.api_version`, client library
+generators **must** include the version in requests sent to the API via
+the HTTP Header `X-Goog-Api-Version`.
+
+Requests which contain the API version, **must** include either an HTTP query
+parameter `$apiVersion` or HTTP header `X-Goog-Api-Version`, but a request
+**must not** contain both.
+
+Generated documentation for a given service **may** include the value of
+`google.api.api_version`, if it exists in the source protos.
+
+## Rationale
+
+The `google.api.api_version` annotation allows services to abide by the schema and
+service behavior at the time the API version was deployed. The format of the
+API version **must** be treated as opaque by clients. Generators and clients **must not**
+interpret the value of `google.api.api_version` beyond the uses mentioned above.
```

Comment:
The Rationale shouldn't contain guidance, that should be in the guidance section above. Some examples of other Rationale sections include https://google.aip.dev/203#rationale and https://google.aip.dev/202#rationale

These basically answer the questions of "why X guidance?" which come up often after N weeks/months in the wild.

I'd recommend three subsections, one for why clients must do this at all, one for why clients need to treat the value as opaque, and one why both query & header cannot be set at the same time